### ISSUE TITLE
request-params doesn't handle "OPTIONS" and "PUT"

### ISSUE THREAD

**Issue by erg** [factor/factor#2106]
```factor
: request-params ( request -- assoc )
    dup method>> {
        { "GET" [ url>> query>> ] }
        { "HEAD" [ url>> query>> ] }
        { "POST" [ post-data>> params>> ] }
    } case ;
```

```
[2019-01-07T19:41:31.354379Z] ERROR do-request: Fall-through in case
[--------------------] ERROR do-request: object "OPTIONS"
[--------------------] ERROR do-request: (U) Quotation: [ set-namestack init-catchstack self quot>> call => stop ]
[--------------------] ERROR do-request: (O) Word: handle-client
[--------------------] ERROR do-request: (O) Method: M\ http-server handle-client*
[--------------------] ERROR do-request: (O) Word: do-request
[--------------------] ERROR do-request: (O) Word: init-request
[--------------------] ERROR do-request: (O) Word: request-params
[--------------------] ERROR do-request: (O) Word: no-case
[--------------------] ERROR do-request: (O) Method: M\ object throw
[--------------------] ERROR do-request: (U) Quotation: [
[--------------------] ERROR do-request:         OBJ-CURRENT-THREAD special-object error-thread set-global
[--------------------] ERROR do-request:         current-continuation => error-continuation set-global
[--------------------] ERROR do-request:         [ original-error set-global ] [ rethrow ] bi
[--------------------] ERROR do-request:     ]
```

```
[2019-01-08T00:57:12.431444Z] ERROR do-request: Fall-through in case
[--------------------] ERROR do-request: object "PUT"
[--------------------] ERROR do-request: (U) Quotation: [ set-namestack init-catchstack self quot>> call => stop ]
[--------------------] ERROR do-request: (O) Word: handle-client
[--------------------] ERROR do-request: (O) Method: M\ http-server handle-client*
[--------------------] ERROR do-request: (O) Word: do-request
[--------------------] ERROR do-request: (O) Word: init-request
[--------------------] ERROR do-request: (O) Word: request-params
[--------------------] ERROR do-request: (O) Word: no-case
[--------------------] ERROR do-request: (O) Method: M\ object throw
[--------------------] ERROR do-request: (U) Quotation: [
[--------------------] ERROR do-request:         OBJ-CURRENT-THREAD special-object error-thread set-global
[--------------------] ERROR do-request:         current-continuation => error-continuation set-global
[--------------------] ERROR do-request:         [ original-error set-global ] [ rethrow ] bi
[--------------------] ERROR do-request:     ]
```

**Comment by erg** [comment]
Maybe this:

```factor
IN: http.server

: request-params ( request -- assoc )
    dup method>> {
        { "GET" [ url>> query>> ] }
        { "HEAD" [ url>> query>> ] }
        { "OPTIONS" [ url>> query>> ] }
        { "POST" [ post-data>> params>> ] }
        { "PUT" [ post-data>> params>> ] }
    } case ;

8888 httpd
"localhost:8888" http-options
"hi" "localhost:8888" http-put

```

**Comment by mrjbq7** [comment]
Fixed in a0fb56ab00c2aca3cfdfc886cd4ed236a5a64f04.
